### ISSUE TITLE
Reg\Register an Operation\Page "Operation Information\Opted-in Operation" - External user can register the operation

### ISSUE THREAD

**Issue by andrea-williams** [bcgov/cas-registration#1952]
#### Dependencies

- Blocked by #1899 
- Blocked by #2133 
- Blocking https://github.com/bcgov/cas-registration/issues/2094

#### Description:

Brief description of the work involved, including any user research context or key decisions.

In Registration workflow mode, [wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=13653-2443&t=HA6SzABkE1DQWPQg-4)

#### Acceptance Criteria:

Given an authenticated external user has selected "Opted-in Operation" as the purpose of registration in the Registration workflow
Then they get an extra step at step 3 (after facility information) called Opt-In Application
And they see [this form
](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=13653-2443&t=eV3D2Yqyi2or0X8R-4)

Given an authenticated external user has selected "Opted-in Operation" as the purpose of registration in the Registration workflow
When they get to the "Operation Information" step of the Registration form
Then they see [this](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=13340-2186&m=dev&t=NI9Z4bjk0RMG6DoE-1)
And when they click the "Save and Continue" button, the data they input is saved to our database

#### Development Checklist:

Create new data model `Opted In Operation`:
- [x] FK reference to operation ID
- [x] json that will have Yes/No answers to the following questions

1. Does this operation have emissions that are attributable for the purposes of section 3 of the Act?
2. Is this operation an electricity import operation?
3. Is this operation a part of a single facility operation or a linear facilities operation?
4. Does this operation have emissions that are attributable for the purposes of section 6 of the Act?
5. Is this operation's primary economic activity classified by the following NAICS codes: 11, 22, or 562?
6. Does this operation produce a regulated product listed in Table 1 of Schedule A.1 of the GGERR?
7. Is this operation capable of fulfilling the obligations of a reporting operation and a regulated operation under the Act and the regulations?
8. Will the operator notify the Director as soon as possible if this operation ceases to meet any of the criteria for the designation of the operation as a reporting operation and a regulated operation?

- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

**Comment by shon-button** [comment]
@andrea-williams 
For your consideration, to enhance clarity and robustness; perhaps, it would be beneficial to reference steps by their specific names rather than their numerical order. For instance, instead of this detail, "When they get to the 4th step of the Registration form," it might be more clear as this detail, "When they get to the Operation Information step of the Registration form."

**Comment by Sepehr-Sobhani** [comment]
@andrea-williams May I move the last AC of this ticket to a separate ticket and block it with this ticket? I want to do this to avoid creating a monster PR.

**Comment by Sepehr-Sobhani** [comment]
@andrea-williams Created this ticket: https://github.com/orgs/bcgov/projects/122/views/2?filterQuery=&pane=issue&itemId=74955567 and I gave it a 3.

**Comment by shon-button** [comment]
@patriciarussellCAS

Reachable at:
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/registration/register-an-operation

**Comment by hannavovk** [comment]
This one doesn't seem to be working in dev - despite selecting "opt-in" in get the "new entrant" workflow. The same happens if I select any other registration purpose. (@Sepehr-Sobhani, @patriciarussellCAS )

**Comment by hannavovk** [comment]
Also, the 'steps' at the top show the "new entrant application" step before any registration purpose is selected. (cc: @Sepehr-Sobhani, @patriciarussellCAS ) 


![Image](https://github.com/user-attachments/assets/b2508de1-b00e-4e7e-bacc-8e473e4ee428)

**Comment by Sepehr-Sobhani** [comment]
> Also, the 'steps' at the top show the "new entrant application" step before any registration purpose is selected. (cc: @Sepehr-Sobhani, @patriciarussellCAS )
> 
> ![Image](https://private-user-images.githubusercontent.com/87211104/365323567-b2508de1-b00e-4e7e-bacc-8e473e4ee428.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2NjUxMTIsIm5iZiI6MTcyNTY2NDgxMiwicGF0aCI6Ii84NzIxMTEwNC8zNjUzMjM1NjctYjI1MDhkZTEtYjAwZS00ZTdlLWJhY2MtOGU0NzNlNGVlNDI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA2VDIzMjAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0NWJmOWRjMzYxY2ZjNmQzNmM1YjZiODczZGEwNTQ5YjAyNmM3ZWRjMjdjNmEzN2JmZmI2MGUzODc0MTZhYTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9wlARQ2MFoAOBCqG6JKLSyZhOa6jA_N24O5i1rw2O8w)

Yes, the registration purpose is hard coded for now and we have captured the tasks to resolve this issue in this ticket: https://github.com/orgs/bcgov/projects/122/views/2?filterQuery=purpose&pane=issue&itemId=76712500

**Comment by patriciarussellCAS** [comment]
Hey @Sepehr-Sobhani could you walk me through this one? 
In dev, when I select opted-in operation for purpose I'm still not seeing additional data fields in the operation information section for opt-ins. 
I may be doing something wrong though!

**Comment by Sepehr-Sobhani** [comment]
> Hey @Sepehr-Sobhani could you walk me through this one? In dev, when I select opted-in operation for purpose I'm still not seeing additional data fields in the operation information section for opt-ins. I may be doing something wrong though!

Hey Patricia! The registration purpose is currently hard coded, so we won't be able to see this in dev until [this](https://github.com/orgs/bcgov/projects/122/views/2?filterQuery=purpose&pane=issue&itemId=76712500) ticket is merged. However, I can guide you through it using my local environment if that works for you?!

**Comment by patriciarussellCAS** [comment]
Thanks Sepehr. #2133 is in pending PO approval...but maybe it hasn't been merged yet? 
Perhaps we can connect on this tomorrow if you're around :)

**Comment by Sepehr-Sobhani** [comment]
Aw, sorry, I didn't notice it had been merged! I tried this feature and was able to get to the 'opted-in' page in Dev.

![Screenshot 2024-09-19 at 3 53 10 PM](https://github.com/user-attachments/assets/55740713-3d9b-459f-9d39-43cfbb9b7e15)

I can guide you through this tomorrow if you want.

**Comment by patriciarussellCAS** [comment]
I was able to review this in Dev and confirm that it works as intended. 

My review notes: 
- Do we need to reject anyone who selects "NO" to any of the questions in the Opt-in Application form? Or do we allow them to continue through to the end of registration? 
- Follow-up question for BA: How/when are they reviewing the answers in the Opt-in Application form?
